src/Sarif/RuleUtilities.cs

```diff
@@ -105,6 +106,12 @@ public static string NormalizeTemplateId(string ruleId, string templateId)
             {
                 templateId = templateId.Substring(ruleId.Length + 1);
             }
+
+            if (!string.IsNullOrEmpty(prefix) && templateId.StartsWith(prefix + "_", StringComparison.Ordinal))
+            {
+                templateId = templateId.Substring(prefix.Length + 1);
```

Comment:
>templateId.Substring(prefix.Length + 1 [](start = 29, length = 38)

you said in the description that the prefix would be inserted after the rule id, but here, you permit the string to be prefixed with the prefix. 

need to be consistent. if we follow your change description, we don't need this new conditional, which is nice

Reply:
The method `NormalizeTemplateId` first strips the `<ruleId>` (if present), and then strips the `<prefix>` (if present). So any of these will work (in the sense that when `NormalizeTemplateId` is finished), only `<templateId>` will remain:

- `<ruleId>_<prefix>_<templateId>`
- `<ruleId>_<templateId>`
- `<prefix>_<templateId>`
- `<templateId>`

Those last two would be useful when the same message is shared by multiple rules (skimmers).